frontend/src/component/hoc/withPageAuth.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { NotFoundPage } from '../pages/NotFoundPage';
+
+export default function(
+  WrappedComponent: React.ComponentType<any>,
+  makeAuthDecision: Promise<boolean>
+) {
+  enum AuthDecision {
+    AUTHORIZED,
+    UNAUTHORIZED,
+    PENDING
+  }
+
+  interface IState {
+    decision: AuthDecision;
+  }
+
+  return class extends React.Component<any, IState> {
+    private isComponentMounted: boolean;
+
+    constructor(props: any) {
+      super(props);
+      this.state = {
+        decision: AuthDecision.PENDING
+      };
+      this.isComponentMounted = false;
+    }
+
+    componentDidMount(): void {
+      this.isComponentMounted = true;
+
+      makeAuthDecision.then(decision => {
+        if (!this.isComponentMounted) {
+          return;
+        }
+        this.setState({
+          decision: decision
+            ? AuthDecision.AUTHORIZED
```

Comment:
Recommend wrapping this into a private method: `toAuthDecision(featureDecision: boolean): AuthDecision`